cmstestsuite/testrunner.py

```diff
@@ -304,6 +304,15 @@ def submit_tests(self, concurrent_submit_and_eval=True):
             self.ps.start("Worker", shard)
         if concurrent_submit_and_eval:
             self.ps.start("EvaluationService", contest=self.contest_id)
+            # We need to wait on ES as otherwise CWS would receive
+            # submissions before it has established a connection with
+            # ES, and this would cause ES not to receive notifications
+            # about new submissions, which would only get picked up at
+            # the following sweep, delaying judging. Moreover a race
+            # condition causes the first sweep of ES not to pick up
+            # some of the submissions that are being committed by CWS
+            # to the DB but that aren't being notified via RPC.
```

Comment:
Can I rephrase?

```
A submission can arrive after ES' first sweep, but before CWS connects
to ES; if so, it will be ignored until the second sweep, making the test
flaky. By waiting for ES to start before submitting, we ensure CWS can
send the notification for all submissions.
```